app/src/ui/add-repository/add-existing-repository.tsx

```diff
@@ -81,18 +103,39 @@ export class AddExistingRepository extends React.Component<IAddExistingRepositor
     this.checkIfPathIsRepository(path)
   }
 
+  private pathExists(directory: string): Promise<boolean> {
+    return new Promise<boolean>((resolve, reject) => {
+      try {
+        FS.stat(directory, (err, exists) => {
+          if (err) {
+            resolve(false)
+          }
+
+          resolve(true)
+        })
+      } catch (e) {
+        resolve(false)
+      }
+    })
+  }
+
   private async checkIfPathIsRepository(path: string) {
-    this.setState({ path, isGitRepository: null })
+    this.setState({ path })
 
     const token = ++this.checkGitRepositoryToken
+    const pathExists = await this.pathExists(path)
+
+    let isRepo = false
 
-    const isRepo = await isGitRepository(this.resolvedPath(path))
+    if (pathExists) {
+      isRepo = await isGitRepository(this.resolvedPath(path))
+    }
 
     // Another path check was requested so don't update state based on the old
     // path.
     if (token !== this.checkGitRepositoryToken) { return }
 
-    this.setState({ path: this.state.path, isGitRepository: isRepo })
+    this.setState({ path: this.state.path, isGitRepository: isRepo, showNonGitRepositoryWarning: !isRepo })
   }
 
   private resolvedPath(path: string): string {
```

Comment:
I think we should make sure that the `resolvedPath` always returns an absolute path. Otherwise we get into some weird edge-cases on Windows.

Perhaps something like

```ts
return Path.resolve('/', untildify(path))
```

Here's one example of path edge-casing on Windows

<img src="https://user-images.githubusercontent.com/634063/26881238-2b2d6d1e-4b97-11e7-8ee3-42d814f528e6.gif" width="400" />

In the first case `c` the path becomes `cwd+'c'` which (on my machine in dev mode) becomes `c:\Users\Markus\Documents\GitHub\Desktop\c` (which doesn't exist).

In the second case (since c: is a weird one on Windows that behaves somewhat as an alias to cwd) of `c:` the path becomes just `cwd` (`c:\Users\Markus\Documents\GitHub\Desktop\`, which does exist).

And in the third case we're doing the same thing as in #1.